addons/binding/org.openhab.binding.hpprinter/src/main/java/org/openhab/binding/hpprinter/internal/handler/HPPrinterHandler.java

```diff
@@ -0,0 +1,223 @@
+            logger.debug("Using SSL for {}", getThing().getUID().toString());
+            HttpsTrustManager.allowAllSSL(); // Ignore certificates as most HP Printer certs are invalid.
+        }
+
+        setDefaultChannelValues(); // Set default channel values
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        if (SUPPORTED_DEVICES.contains(getThing().getThingTypeUID())) {
+            this.printer = new DefaultRequestHandler(config, getThing());
+        }
+
+        this.printer.setEventListener(this);
+
+        if (config.ipAddress != "") {
+            // Example for background initialization:
+            scheduler.execute(() -> {
+                boolean thingReachable = this.printer.checkIfOnline(); // <background task with long running
+                                                                       // initialization
+                // when done do:
+                if (thingReachable) {
+                    startBackgroundSchedules();
+                } else {
+                    thingGoneOffline(true);
+                }
+            });
+
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "You must set an IP Address");
+        }
+    }
+
+    private void setDefaultChannelValues() {
+        updateState(new ChannelUID(getThing().getUID(), CGROUP_CONTROL, CHANNEL_REBOOT), OnOffType.OFF);
+    }
+
+    // Run this when the printer has gone offline.
+    public void thingGoneOffline(boolean rescanCheck) {
+        logger.debug("HP Printer {} has gone offline", getThing().getUID().toString());
+        stopBackgroundSchedules();
+        // Set thing to offline and status channel to offline.
+        updateStatus(ThingStatus.OFFLINE);
+        updateState(new ChannelUID(getThing().getUID(), CGROUP_STATUS, CHANNEL_STATUS), new StringType("Offline"));
+        if (rescanCheck) {
+            thingOnlineCheck();
+        }
+    }
+
+    public void thingBackOnline() {
+        startBackgroundSchedules();
+        if (onlineCheckScheduler != null) {
+            onlineCheckScheduler.cancel(false);
+        }
+    }
+
+    // Check periodically to see if the printer has come back online.
+    public void thingOnlineCheck() {
+        onlineCheckScheduler = scheduler.scheduleWithFixedDelay(() -> {
+            if (this.printer.checkIfOnline()) {
+                thingBackOnline();
+            }
+        }, 0, onlineCheckPeriod, TimeUnit.SECONDS);
+    }
+
+    // Stops updating of status and usage information.
+    public void stopBackgroundSchedules() {
+        if (usageScheduler != null) {
+            usageScheduler.cancel(true);
+        }
+        if (statusScheduler != null) {
+            statusScheduler.cancel(true);
+        }
+    }
+
+    // Starts updating of status and usage information.
+    public void startBackgroundSchedules() {
+        updateStatus(ThingStatus.ONLINE);
+
+        statusScheduler = scheduler.scheduleWithFixedDelay(() -> {

```

Comment:
Should we check the status every 4 seconds, if someone uses this in an office with 100 printers this could become messy

Reply:
Now a separate configuration option.